File: src/main/java/ru/serdtsev/homemoney/dto/Turnover.java
package ru.serdtsev.homemoney.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import ru.serdtsev.homemoney.account.AccountType;

import javax.annotation.Nonnull;
import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Обороты за день по типу счета
 */
@Data
@AllArgsConstructor
@RequiredArgsConstructor
@EqualsAndHashCode(of = {"operDate", "accountType"})
public class Turnover {
  @Nonnull private final LocalDate operDate;
  @Nonnull private final AccountType accountType;
  /** Сумма оборотов со знаком */
  @Nonnull private BigDecimal amount = BigDecimal.ZERO;

  public void plus(BigDecimal value) {
    amount = amount.add(value);
  }


}

File: src/main/java/ru/serdtsev/homemoney/dto/Turnover.kt
package ru.serdtsev.homemoney.dto

import java.math.BigDecimal
import java.sql.Date

class Turnover(var trnDate: Date? = null, var fromAccType: Account.Type? = null, var toAccType: Account.Type? = null) {
  var amount = BigDecimal.ZERO

  override fun equals(other: Any?): Boolean{
    if (this === other) return true
    if (other?.javaClass != javaClass) return false

    other as Turnover

    if (trnDate != other.trnDate) return false
    if (fromAccType != other.fromAccType) return false
    if (toAccType != other.toAccType) return false

    return true
  }

  override fun hashCode(): Int{
    var result = trnDate?.hashCode() ?: 0
    result += 31 * result + (fromAccType?.hashCode() ?: 0)
    result += 31 * result + (toAccType?.hashCode() ?: 0)
    return result
  }
}
